### ISSUE TITLE
Some lightweight themes aren't displayed correctly

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-frontend#1964]
### Describe the problem and steps to reproduce it:
1. Go to https://addons-dev.allizom.org/en-US/android/search/?q=nature
2. Scroll down and look for Nature Romance (should be third).
3. Click on it.

### What happened?
Theme isn't shown in mobile view. The search result is blank, and the listing page doesn't show the image.

### What did you expect to happen?
Theme should show up like in desktop front-end.

**Comment by jvillalobos** [comment]
May be a caching thing. Visiting the theme page directly looks right. Reloading the listing page after navigating from the search page also makes the image appear.

**Comment by ValentinaPC** [comment]
Check out #1785 .
Thanks!

**Comment by muffinresearch** [comment]
The specific details for this one are:

This is the data via search:

```json
{
  "theme_data": {
    "category": "Nature",
    "iconURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/preview_small.jpg?1374358516",
    "headerURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/header.png?1374358516",
    "name": "Nature Romance",
    "author": "Mirian Sousa",
    "footer": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/footer.png?1374358516",
    "previewURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/preview_large.jpg?1374358516",
    "updateURL": "https:\/\/versioncheck.addons.mozilla.org\/en-US\/themes\/update-check\/443600",
    "accentcolor": "#001842",
    "header": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/header.png?1374358516",
    "version": "1.0",
    "footerURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/footer.png?1374358516",
    "detailURL": "https:\/\/addons-dev.allizom.org\/en-US\/firefox\/addon\/nature-romance\/",
    "textcolor": "#f7f2f2",
    "id": "443600",
    "description": "None"
  }
}
```

This is the data for the listing on the details page:

```json
{
  "theme_data": {
    "category": "Nature",
    "iconURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/icon.png?1374358516",
    "headerURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/header.png?1374358516",
    "name": "Nature Romance",
    "author": "Mirian Sousa",
    "footer": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/footer.png?1374358516",
    "previewURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/preview.png?1374358516",
    "updateURL": "https:\/\/versioncheck.addons.mozilla.org\/en-US\/themes\/update-check\/443600",
    "accentcolor": "#001842",
    "header": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/header.png?1374358516",
    "version": "1.0",
    "footerURL": "https:\/\/addons-dev-cdn.allizom.org\/user-media\/addons\/443600\/footer.png?1374358516",
    "detailURL": "https:\/\/addons-dev.allizom.org\/en-US\/firefox\/addon\/nature-romance\/",
    "textcolor": "#f7f2f2",
    "id": "443600",
    "description": ""
  }
}
```

The data for the details page and the search results are completely different (particularly note the preview URL). It's probably due to data import that these are different. One check might be to re-index the add-ons which might solve the data discrepancy.

**Comment by diox** [comment]
I don't think a reindex would solve this. There is a weird hack to support both old and new Personas in the model, and we index things to support that hack but it needs some tweaking, I'm on it.

**Comment by ValentinaPC** [comment]
This was verified as fixed on #1785.
